src/executeJar.ts

```diff
@@ -240,6 +241,62 @@ export async function loadContextsFromChangelogFile(
       reject(error as Error);
     }
   });
+
+  /**
+   * Sets the context for all files that point to the same changelog file.
+   *
+   * @param contexts - the contexts that were loaded from the changelog file
+   * @param changelogFile - the absolute path to the changelog file
+   */
+  function setContextForCache(contexts: string[], changelogFile: string): void {
+    readLiquibaseConfigurationNames()
+      .then((names) => {
+        if (!names) {
+          return;
+        }
+        for (const configurationName of names) {
+          getPathOfConfiguration(configurationName)
+            .then((propertyFile) => {
+              if (!propertyFile) {
+                return;
+              }
+
+              const data = readFullValues(configurationName, propertyFile, {
+                defaultDatabaseForConfiguration: getDefaultDatabaseForConfiguration(),
+                liquibaseDirectoryInProject: getLiquibaseFolder(),
+                customDrivers: getCustomDrivers(),
+              });
+
+              const previousSelectedContexts: ContextSelection = cacheHandler.readContexts(
+                propertyFile,
+                data.changelogFile
+              );
+
+              if (data.changelogFile.includes(path.relative(getWorkFolder(), changelogFile))) {
+                // save the loaded context into the cache
+                cacheHandler.saveContexts(propertyFile, changelogFile, {
+                  loadedContexts: contexts,
+                  selectedContexts: previousSelectedContexts.selectedContexts ?? [""],
```

Comment:
Unfortunately, this does still not work correctly.

1. After loading the contexts, an entry for all given configurations are stored. This should not matter, because it has no data 
2. When you are loading the first time contexts from a changelog that is not in the cache for this configuration, but for another properties file, 
 -  it does still remove the selected contexts from the previous loaded contexts
 - it does not store the loaded contexts for the new configuration

See more details in teams video.